hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -1752,6 +1752,21 @@ class IRSuite extends HailSuite {
     assertEvalsTo(LowerBoundOnOrderedCollection(dwoutna, NA(TInt32), onKey = true), 2)
   }
 
+  @Test def testStreamLen(): Unit = {
+    val a = StreamLen(MakeStream(Seq(I32(3), NA(TInt32), I32(7)), TStream(TInt32)))
+    assertEvalsTo(a, 3)
+
+    val missing = StreamLen(NA(TStream(TInt64)))
+    assertEvalsTo(missing, null)
+
+    val range1 = StreamLen(StreamRange(1, 11, 1))
+    assertEvalsTo(range1, 10)
+    val range2 = StreamLen(StreamRange(20, 40, 2))
+    assertEvalsTo(range2, 10)
+    val range3 = StreamLen(StreamRange(-10, 5, 1))
+    assertEvalsTo(range3, 15)
```

Comment:
I would at least add something like a `StreamLen(StreamFilter(...))`, that doesn't use the known-length path.

Another good idea would be to add test cases for each stream node that has a known length, to make sure the length computations are correct more than testing `StreamLen` itself. Those might be better in the test functions for the respective stream nodes. I think right now the list is `StreamRange`, `ToStream`, `MakeStream`, `StreamTake`, `StreamDrop`, `StreamGrouped`, `StreamMap`, `StreamZip`, `If`, `StreamScan`, `RunAggScan`, `StreamLeftJoinDistinct`. But that's a lot, we could also just make a task to add those later.

Reply:
Most of these have tests now.

Reply:
Thanks!